### PR TITLE
ZCS-643: Universal UI- Styling changes for mail search left side bar

### PR DESCRIPTION
**Pull request by kalpeshhpatel**
Changes:
* ZmSearchResultsFilterPanel.js:
  * Change `overflow-y:auto` for container
  * Added new class for search query operators.
* Updated `dwt/Widgets.template` and `share/Widgets.template` as per requirement
* Added/updated css rules in `zm.css` and `skin.css`
* Added/updated properties in `skin.properties`
* Added/updated properties in `ZmMsg.properties`

https://jira.corp.synacor.com/browse/ZCS-643

### CODE REVIEW

File: WebRoot/css/zm.css [hunk 1]
```diff
@@ -4594,14 +4609,20 @@ HTML>BODY .appt-selected .appt_allday_body
 	@ActiveCursor@
 }
 
+.ZCheckboxTable,
+.ZRadioButtonTable {
+	height: 100%;
+}
+
+
```
Comment (tarangkhandelwal) [inline WebRoot/css/zm.css]: extra space?

File: WebRoot/css/zm.css [hunk 2]
```diff
@@ -4594,14 +4609,20 @@ HTML>BODY .appt-selected .appt_allday_body
...
 .ZCheckboxTable>TBODY>TR>TD,
 .ZRadioButtonTable>TBODY>TR>TD {
 	padding:3px 0px;
 }
 
 .ZCheckboxTable INPUT[type="checkbox"] {
 	padding:0px;
-	margin:0px 3px;
+	margin: 0;
```
Comment (tarangkhandelwal) [inline WebRoot/css/zm.css]: small, space before 0 ?

File: WebRoot/templates/share/Widgets.template [hunk 3]
```diff
@@ -96,9 +96,16 @@
 			<td>
 			    <table role="presentation">
 			        <tr id='${id}_items'>
-                        <td id='${id}_inputFieldCell' class='ZmSearchResultInputField'></td>
-                        <td id='${id}_searchButton'   class='ZmSearchResultButton'></td>
-                        <td id='${id}_saveButton'     class='ZmSearchResultButton'></td>
+                        <td class='ZmSearchResultInputField'>
+							<div id='${id}_inputFieldCell'></div>
+							<div class="ZmSearchResultInputFieldIcon">
```
Comment (tarangkhandelwal) [inline WebRoot/templates/share/Widgets.template]: indentation?

Comment (kalpeshhpatel) [inline WebRoot/templates/share/Widgets.template]: Fixed: https://github.com/Zimbra/zm-web-client/pull/8/files#diff-709faf81230ea6ed87cb3202bf97e3c8R98

File: WebRoot/css/zm.css (hunk 2 above)
Comment (kalpeshhpatel) [inline WebRoot/css/zm.css]: Fixed: https://github.com/Zimbra/zm-web-client/pull/8/files#diff-33f9c57f22cd743824b2918132c5d73bR4625

File: WebRoot/css/zm.css (hunk 1 above)
Comment (kalpeshhpatel) [inline WebRoot/css/zm.css]: Fixed: https://github.com/Zimbra/zm-web-client/pull/8/files#diff-33f9c57f22cd743824b2918132c5d73bR4613

File: WebRoot/skins/_base/base4/skin.properties [hunk 4]
```diff
@@ -1670,3 +1670,32 @@ yellowgreen =           #9acd32
...
+######################################################################
+# SEARCH TOOLBAR 
+######################################################################
+
+SearchResultToolbar             = background-color:@AppC@; @BottomSeparator@; padding: 0 @SkinWrapperPadding@; box-sizing: border-box; 
+SearchResultInputField          = padding: 0 @SkinWrapperPadding@; width:100%; position: relative;
+SearchResultInputFieldIcon      = position: absolute; top: 50%; margin-top: -10px; right: 24px;
+SearchResultText                = @Text-light@ @FontSize-slightly-bigger@ white-space:nowrap; @AlignRight@ vertical-align:middle; width:15%; min-width:200px;
+SearchResultButton              = display: table-cell;
+SearchFilterContainer           = background:@white@; @roundCorners(2px)@; border-color: @PanelBorderColor@;
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: instead of white can we use AppC ?

File: WebRoot/skins/_base/base4/skin.properties [hunk 5]
```diff
@@ -1670,3 +1670,32 @@ yellowgreen =           #9acd32
...
+SearchFilter                    = height: 3rem; line-height: 2.8rem; padding: 4px 4px 0;
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: for clarity better to use 2/4 values in padding field

File: WebRoot/skins/_base/base4/skin.properties [hunk 6]
```diff
@@ -1670,3 +1670,32 @@ yellowgreen =           #9acd32
 VeilOverlayC1       =   @AltC@
 VeilOverlayC2       =   @darken(AltC,20)@
 VeilOverlayGrad     =   @grad(VeilOverlayC1,VeilOverlayC2,linear-vertical)@
+
+######################################################################
+# SEARCH TOOLBAR 
+######################################################################
+
+SearchResultToolbar             = background-color:@AppC@; @BottomSeparator@; padding: 0 @SkinWrapperPadding@; box-sizing: border-box; 
+SearchResultInputField          = padding: 0 @SkinWrapperPadding@; width:100%; position: relative;
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: @FullWidth@

File: WebRoot/skins/_base/base4/skin.properties [hunk 7]
```diff
@@ -1670,3 +1670,32 @@ yellowgreen =           #9acd32
...
+SearchResultInputFieldIcon      = position: absolute; top: 50%; margin-top: -10px; right: 24px;
+SearchResultText                = @Text-light@ @FontSize-slightly-bigger@ white-space:nowrap; @AlignRight@ vertical-align:middle; width:15%; min-width:200px;
+SearchResultButton              = display: table-cell;
+SearchFilterContainer           = background:@white@; @roundCorners(2px)@; border-color: @PanelBorderColor@;
+SearchFilter                    = height: 3rem; line-height: 2.8rem; padding: 4px 4px 0;
+SearchFilterPanel               = @OverviewTreeContainer@; border:0; width:100%; height:100%;
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: @FullSize@

File: WebRoot/skins/_base/base4/skin.properties [hunk 8]
```diff
@@ -1670,3 +1670,32 @@ yellowgreen =           #9acd32
...
+SearchToolbarHeader             = @TreeHeaderContainer@; @TreeHeaderText@ @TopSeparator@; height: auto; line-height: 3rem; padding:0 @SkinWrapperPadding@;
+SearchToolbarFilter             = padding:0 @SkinWrapperPadding@; height: 3rem;
+SearchToolbarFilterText         = @FontSize-slightly-big@; fill: @PrimaryBrandingColor@
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: property name is FilterText and we are defining fill property in there, isn't this confusing?

File: WebRoot/templates/share/Widgets.template [hunk 9]
```diff
@@ -92,15 +92,23 @@
...
-                        <td id='${id}_searchButton'   class='ZmSearchResultButton'></td>
-                        <td id='${id}_saveButton'     class='ZmSearchResultButton'></td>
-			        </tr>
-			    </table>
+				<table role="presentation">
+					<tr id='${id}_items'>
+						<td class='ZmSearchResultInputField'>
+							<div id='${id}_inputFieldCell'></div>
+							<div class="ZmSearchResultInputFieldIcon">
+								<$= AjxImg.getImageHtml( "Search2") $>
```
Comment (tarangkhandelwal) [inline WebRoot/templates/share/Widgets.template]: space before "Search2"

File: WebRoot/skins/_base/base4/skin.properties (hunk 4 above)
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: I think `InputBgColor` would be more appropriate. 
Updated: https://github.com/Zimbra/zm-web-client/pull/8/files#diff-16ae02866e4f63813de9a7825a1057e3R1683

File: WebRoot/skins/_base/base4/skin.properties (hunk 5 above)
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: Done: https://github.com/Zimbra/zm-web-client/pull/8/files#diff-16ae02866e4f63813de9a7825a1057e3R1684

File: WebRoot/skins/_base/base4/skin.properties (hunk 6 above)
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: Done: https://github.com/Zimbra/zm-web-client/pull/8/files#diff-16ae02866e4f63813de9a7825a1057e3R1679

File: WebRoot/skins/_base/base4/skin.properties (hunk 7 above)
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: Done: https://github.com/Zimbra/zm-web-client/pull/8/files#diff-16ae02866e4f63813de9a7825a1057e3R1685

File: WebRoot/skins/_base/base4/skin.properties (hunk 8 above)
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: `fill` is for targeting SVG icon in that container, and I don't feel need to have other specific property for that. However for readability purpose I have moved `fill` property to `SearchToolbarFilter` property and will work as before.

Thanks for it.

File: WebRoot/templates/share/Widgets.template (hunk 9 above)
Comment (kalpeshhpatel) [inline WebRoot/templates/share/Widgets.template]: Done: https://github.com/Zimbra/zm-web-client/pull/8/files#diff-709faf81230ea6ed87cb3202bf97e3c8R103

File: WebRoot/messages/ZmMsg.properties [hunk 10]
```diff
@@ -3007,10 +3007,10 @@ searchNotes = Notes
 searchPersonalContacts = Search Personal Contacts
 searchPersonalSharedContacts = Personal and Shared Contacts
 searchResults = Search results
-searchResultsEnterSearch = Search for:
+searchResultsEnterSearch = Search for
 # {0} = number of results, {1} = possible plus sign indicating more results
-searchResultsLabel = Found {0}{1} {0,choice,1#result|2#results} for:
-searchResultsLabelNone = Found no results for:
+searchResultsLabel = {0}{1} {0,choice,1#Result|2#Results} for
```
Comment (silentsakky) [inline WebRoot/messages/ZmMsg.properties]: better to not change it as it is shown to users, and was like this before

**Pull request merged**